team/tests/org.eclipse.compare.tests/src/org/eclipse/compare/tests/SimpleIgnoreWhitespaceContributor.java

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.compare.tests;
+
+import java.util.Map.Entry;
+import java.util.TreeMap;
+
+import org.eclipse.compare.contentmergeviewer.IIgnoreWhitespaceContributor;
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.IDocument;
+
+public class SimpleIgnoreWhitespaceContributor implements IIgnoreWhitespaceContributor {
+
+	private final IDocument document;
+	private TreeMap<Integer /* start offset */, Integer /* end offset */> literalsByOffset = new TreeMap<>();
+
+	public SimpleIgnoreWhitespaceContributor(IDocument document) {
+		this.document = document;
+		scanAndCreateLiteralsByOffset();
+	}
+
+	private void scanAndCreateLiteralsByOffset() {
+		String s = document.get();
+		boolean inStringLiteral = false;
+		int start = 0, end = 0;
+		for (int i = 0; i < s.length(); i++) {
+			char c = s.charAt(i);
+			if (c == '\\' && inStringLiteral) {
+				i++;
+			} else if (c == '"') {
+				inStringLiteral = !inStringLiteral;
+				if (inStringLiteral) {
+					start = i;
+				} else {
+					end = i;
+					literalsByOffset.put(start, end);
+				}
+			}
+		}
+	}
+
+	@Override
+	public boolean isIgnoredWhitespace(int lineNumber, int columnNumber) {
+		try {
+			int offset = document.getLineOffset(lineNumber) + columnNumber;
+			Entry<Integer, Integer> entry = literalsByOffset.floorEntry(offset);
+			if (entry != null) {
+				int start = entry.getKey();
+				int end = entry.getValue();
+				if (offset >= start && offset <= end) {
+					return false; // part of literal - whitespace cannot be ignored
+				}
+			}
+		} catch (BadLocationException e) { // NOPMD
```

Comment:
Maybe you can add nicer exception handling.
I am not sure if Platform uses PMD

Reply:
this is a common issue. What should we do if a BadLocationException is thrown? Here in this case, we cannot calculate the offset and therefore are not able to check if given whitespace is part of a literal. 
Is it worth to stop the calculation of the differences? I am not quite sure.

I checked how the compare plugin handles the BadLocationExceptions:
- most occurrences use "// silently ignored" comment
```
 } catch (BadLocationException x) {
   // silently ignored
}
```
- at some places System.out.println is used if debug is enabled. I don't thinks that this approach should be preferred.
```
} catch (BadLocationException ex) {
  // silently ignored
  if (TextMergeViewer.DEBUG) System.out.println("BadLocationException: " + ex);	//$NON-NLS-1$
}
```
- at one place I found a log call which might be a good alternative instead of ignoring it. I therefore have chosen this solution. But, there are maybe two reasons not to use it:
1. users might find the log entry and create tickets stating that this is an error which must be fixed. Maybe they are right. I am not quite sure.
2. this code here is only test code trying to explain how an implementation of IIgnoreWhitespaceContributor could be realized. It is called by a unit test. If you run the test as unit test, ComparePlugin is not initialized and CompareUIPlugin.log will run into a NullPointerException.
```
} catch (BadLocationException ex) {
  CompareUIPlugin.log(ex);
}
```

So, in the end, I don't care. Please tell me your preferred solution.